pkg/util/ipvs/ipvs.go

```diff
@@ -139,3 +147,23 @@ func GetRequiredIPVSModules(kernelVersion *version.Version) []string {
 func IsRsGracefulTerminationNeeded(proto string) bool {
 	return !strings.EqualFold(proto, "UDP") && !strings.EqualFold(proto, "SCTP")
 }
+
+// GetRequiredSchedulerModules returns the required ipvs scheduler module for configured scheduler
+func GetRequiredSchedulerModules(scheduler string) string {
```

Comment:
I don't think we want to expand the list of "required" scheduling modules, but we may want to check a specific module only if it was passed into kube-proxy via the --ipvs-scheduler flag.

Reply:
Do you mean this should be renamed GetSchedulerModules?

I am checking the scheduler config [here](https://github.com/kubernetes/kubernetes/pull/93040/files#diff-934760ea1a4e195e49bcd10625cf4e46R125) and it should be set to round-robin by the [DefaultScheduler](https://github.com/kubernetes/kubernetes/blob/86096addb146c740ace72daf1291f81ef55490f2/pkg/proxy/ipvs/proxier.go#L433-L436) if not passed by the administrator.

Reply:
Ah I see, this is returning a single module, thought it was returning a list.

Reply:
Do you think `GetRequiredSchedulerModules` should still have a predefined list of schedulers or should we always check for the kernel module `ip_vs_<scheduler>` and error if it doesn't exist?

Reply:
I'm indifferent to the list vs interpolation. I can change it to that if you'd like.

Reply:
For any non-standard scheduler, I think we should interpolate the module name (ip_vs_<scheduler>) and log a warning if it's not installed with a fallback back to RR.